include/aspect/structured_data.h

```diff
@@ -87,14 +87,41 @@ namespace aspect
          *
          * The data in @p data_table consists of a Table for each of the @p n_components components.
          *
-         * The last two arguments are rvalue references, and the function will
-         * move the data so provided into another storage location. In other
-         * words, after the call, the variables passed as the last two
-         * arguments may be empty or otherwise altered.
+         * The `coordinate_values` and `data_table` arguments are rvalue references,
+         * and the function will move the data so provided into another storage
+         * location. In other words, after the call, the variables passed as the
+         * last two arguments may be empty or otherwise altered.
+         *
+        #if DEAL_II_VERSION_GTE(9,3,0)
+         * If ASPECT is built on deal.II version 9.3 or higher, this class
+         * is able to share data between processes located on the same
+         * machine. In this case, if the last argument, `root_process` is
+         * set to anything other than `numbers::invalid_unsigned_int`, then only the
+         * indicated root process within the given MPI communicator needs to
+         * pass in valid arguments whereas on all other processes the values
+         * of `column_names`, `coordinate_values`, and `data_table` are
+         * ignored. Instead, the indicated root process will make sure that
+         * the other processes obtain valid data, for example by sharing
+         * data tables in shared memory spaces. This reduces both the
+         * computational effort in reading data from disk and parsing it on
+         * all processes, and can vastly reduce the amount of memory required
+         * on machines where many processor cores have access to shared memory
+         * resources.
+         *
+         * If `root_process` equals `numbers::invalid_unsigned_int`, then
+         * every process needs to pass data for all arguments and the
+         * `mpi_communicator` argument is ignored.
+        #else
+         * If ASPECT is built on a version of deal.II older than 9.3,
+         * then all processes need to pass in valid values for the first
+         * three arguments and the last two arguments are ignored.
+        #endif
          */
         void reinit(const std::vector<std::string> &column_names,
                     std::vector<std::vector<double>> &&coordinate_values,
-                    std::vector<Table<dim,double> > &&data_table);
+                    std::vector<Table<dim,double> > &&data_table,
+                    const MPI_Comm &mpi_communicator,
+                    const unsigned int root_process);
```

Comment:
could you provide defaults for these parameters? That way we stay backward compatible with user plugins that use this class.

Reply:
Hm, what should the default values be? I could give `MPI_COMM_SELF` to the former and the special value `numbers::invalid_unsigned_int` to the latter. If a user just calls the old version with both arguments defaulted, that works. If a user calls the new version with both arguments given, that also works. This scheme also seems to work if someone uses a provided communicator and defaults the last argument -- in that case, the communicator is simply ignored, which is also what is documented. 

So I think I just talked myself through doing just that! :-)